articles/basics-of-the-object-model/index.md

```diff
@@ -0,0 +1,106 @@
+
+The primary purpose of OOA is identifying the application domain and gathering the requirements of the system. It mainly focuses on what the system will do rather than how it will do it.
+
+There are three OOA components; Object Modelling, Dynamic Modelling, and Functional Modelling.
+
+- Object Modelling: Objects are instances of a class.
+
+Object Modelling develops the static formation of the system concerning the object.
+
+It recognizes the objects and the relationship between objects and classes. It identifies the attributes and functions of each class.
+
+- [Dynamic Modelling](https://www.wisdomjobs.com/e-university/object-oriented-analysis-and-design-tutorial-2107/ooad-dynamic-modeling-26532.html): It explains how objects respond to events
+- [Functional Modelling](https://www.tutorialspoint.com/object_oriented_analysis_design/ooad_functional_modeling.htm#:~:text=Functional%20Modelling%20gives%20the%20process,Data%20Flow%20Diagrams%20(DFDs).) indicates the processes executed in an object and how data change when it moves to objects.
+
+#### Object-Oriented Design (OOD)
+In OOD, a system is a group of objects. It involves designing the objects, classes, and relationships between classes.
+
+Terms in Object-Oriented Design:
+
+- Instance Variables: Data attributes that are specific to an object.
+- Instance method: Operations that are specific to an object.
+- Class Variables: Data attributes that are not specific to an object.
+- Class Method: Operations that are not specific to an object.
+- Constructors: A unique method that creates and initializes the objects for a class to perform some operations.
+
+### Elements of the Object Model
+Here are the significant elements of the Object Model.
+
+#### Abstraction
+Abstraction reduces complexity. It comes about from the recognition of similarities between objects. Abstraction takes place when the system stress details that are important to the user. It focuses mostly on the outside view of the object. Data is abstracted when protected by a set of methods, and only those methods can access data in the object.
+
+#### Encapsulation
+Encapsulation is achieved through information hiding or data hiding to reduce complexity and increase reusability. The user cannot see the inside of an object or a class, but the object can be accessed by calling the object's methods. Encapsulation and Abstraction are complementary concepts. In Abstraction, the system focus on object behavior and functionality. Encapsulation focus on implementation that gives rise to action.
+
+#### Hierarchy
+Different properties and functions form a class of the hierarchy. Through hierarchy, a class can be composed of inter-related sub-classes, which can have their sub-classes until the smallest level of components is reached. Upper-most classes more general, and classes at the base are more specific.
```

Comment:
Could you please explain this concept further. It seems difficult to understand at a glance.